src/irr/asset/CImageLoaderTGA.cpp

```diff
@@ -165,54 +189,52 @@ asset::IAsset* CImageLoaderTGA::loadAsset(io::IReadFile* _file, const asset::IAs
 
 	uint32_t nullOffset[3] = {0,0,0};
 	uint32_t imageSize[3] = {header.ImageWidth,header.ImageHeight,1};
-
+	bool flip = !((header.ImageDescriptor & 0x20) == 0);
+	
 	switch(header.PixelDepth)
 	{
 		case 8:
 			{
-				if (header.ImageType == 3) // Grayscale
+				if (header.ImageType != 3)
 				{
-					image = new asset::CImageData(NULL,nullOffset,imageSize,0,asset::EF_R8G8B8_UNORM);
-					if (image)
-						video::CColorConverter::convert8BitTo24Bit((uint8_t*)data,
-							(uint8_t*)image->getData(),
-							header.ImageWidth,header.ImageHeight,
-							0, 0, (header.ImageDescriptor&0x20)==0);
-				}
-				else
-				{
-					image = new asset::CImageData(NULL,nullOffset,imageSize,0, asset::EF_A1R5G5B5_UNORM_PACK16);
-					if (image)
-						video::CColorConverter::convert8BitTo16Bit((uint8_t*)data,
-							(int16_t*)image->getData(),
-							header.ImageWidth,header.ImageHeight,
-							(int32_t*) palette, 0,
-							(header.ImageDescriptor&0x20)==0);
+					os::Printer::log("Loading 8-bit non-grayscale is NOT supported.", ELL_ERROR);
+					if (palette) delete [] palette;
+					if (data)    delete [] data;
+					
+					return nullptr;
 				}
+				
+				image = new asset::CImageData(NULL,nullOffset,imageSize,0,asset::EF_R8_UNORM);
```

Comment:
make sure that TGA does not mandate by the spec for greyscale images to also be gamma-encoded